wordpress-mobile/WordPress-Android
# Issue/4972 pending drafts backoff refactor

Fixes #4972

This PR goes with the architecture idea proposed by @maxme and thus ditches the intertwined polling process we had to check for pending drafts.

As soon as you exit a post draft by pressing back without publishing it, this PR creates a +1 day, +1 week, +1 month reminders, with different messages randomly chosen as follows:

- 1 day:

```
You drafted “post title” yesterday. Don’t forget to publish it!
Did you know that “post title” is still a draft? Publish away!
```

- 1 week:

```
Your draft, “post title” awaits you — be sure to publish it!
“Post title” remains a draft. Remember to publish it!
```

- 1 month

```You drafted “post title” X days ago. Your audience awaits it!
Don’t leave it hanging! “Post title” is waiting to be published.
```



To test:
1. Login to the app, and go to Blog Posts
2. Start a new draft by pressing the + floating button
3. write something
4. press back
5. change the device date setting to +1 day, and a new notification should appear. Check that it is one of the +1 type cited above. If you tap EDIT it take you to the draft. Tapping IGNORE will dismiss it.
6. change the device date to +1 week. If the previous notification wasn't dismissed, you'll notice the text for it will have changed to one of the types +1 week cited above. If you dismissed it, a new notification appears, with either 1 of the 2 texts cited above.
7. change the device date to +1 month, you should see a new notification (or if it was there, the existing notification text changes) to the month-version text as cited above.

cc @maxme for review, and cc'ing @daniloercoli as well
